src/System.Management.Automation/security/CredentialParameter.cs

```diff
@@ -59,7 +59,7 @@ public override object Transform(EngineIntrinsics engineIntrinsics, object input
                     // throw an exception
                     if (userName == null)
                     {
-                        throw new PSArgumentException("userName");
+                        throw new PSArgumentException(nameof(inputData), "userName is null");
```

Comment:
The same.

Reply:
Not in this case, because `userName` may be null when the argument `inputData` is not null, if `FromObjectAs` fails.

Reply:
I think about `throw new PSArgumentNullException(nameof(inputData));`

Reply:
No, because `inputData` is not null here this would be incorrect use of `PSArgumentNullException`:

https://github.com/PowerShell/PowerShell/blob/361ddb64db955ea6b21dfb91eb85c2e417be501f/src/System.Management.Automation/security/CredentialParameter.cs#L43